tests/cross/src/test/scala/tests/pc/PcDefinitionSuite.scala

```diff
@@ -351,4 +360,26 @@ class PcDefinitionSuite extends BasePcDefinitionSuite {
        |}
        |""".stripMargin
   )
+
+  check(
+    "synthetic-definition-case-class",
+    """|
+       |class Main {
+       |  case class <<>>User(name: String, age: Int)
+       |  def hello(u: User): Unit = ()
+       |  hello(Us@@er())
+       |}
+       |""".stripMargin
+  )
+
+  check(
+    "synthetic-definition-class-constructor",
+    """|
+       |class Main {
+       |  <<class User(name: String, age: Int)>>
```

Comment:
position isn't perfect. 
It would be nice if it would point at least at the beginning of `User`

Reply:
Agreed, done